src/components/LoginModal/LoginModal.js

```diff
@@ -0,0 +1,101 @@
+import React, { useState, useEffect } from "react";
+import ModalWithForm from "../ModalWithForm/ModalWithForm";
+
+const LoginModal = ({
+  handleCloseModal,
+  onLogin,
+  isOpen,
+  onSwitch,
+  isPasswordError,
+}) => {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+
+  const [isEmailValid, setIsEmailValid] = useState(false);
+  const [isPasswordValid, setIsPasswordValid] = useState(false);
+
+  const [isButtonEnabled, setIsButtonEnabled] = useState(false);
+
+  const handleEmailChange = (e) => {
+    const isEmailValid = e.target.validity.valid;
+    setIsEmailValid(isEmailValid);
+    setEmail(e.target.value);
+    setIsButtonEnabled(isEmailValid && isPasswordValid);
+  };
+
+  const handlePasswordChange = (e) => {
+    const isPasswordValid = e.target.validity.valid;
+    setIsPasswordValid(isPasswordValid);
+    setPassword(e.target.value);
+    setIsButtonEnabled(isEmailValid && isPasswordValid);
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    onLogin({ email, password });
+  };
+
+  useEffect(() => {
+    if (isOpen) {
+      setEmail("");
+      setPassword("");
+      setIsEmailValid(false);
+      setIsPasswordValid(false);
+      setIsButtonEnabled(false);
+    }
+  }, [isOpen]);
+
+  return (
+    <ModalWithForm
+      title="Log In"
+      submitButtonText="Log In"
+      onClose={handleCloseModal}
+      isOpen={isOpen}
+      onSubmit={handleSubmit}
+      isEnable={isButtonEnabled}
+      switchButtonText="or Sign Up"
+      onClick={onSwitch}
+    >
+      <div className="form">
+        <div>
+          <label className="form__label">
+            <p className="form__title">Email</p>
+            <input
+              type="email"
+              className="form__input"
+              name="email"
+              placeholder="Email"
+              required
+              value={email}
+              onChange={handleEmailChange}
+            />
+          </label>
+        </div>
+        <div>
+          <label className="form__label">
+            <p
+              className={`form__title ${
+                isPasswordError ? "form__title-error" : ""
+              }`}
+            >
+              {isPasswordError ? "Incorrect password" : "Password"}
+            </p>
+            <input
+              type="password"
+              className={`form__input ${
+                isPasswordError ? "form__input-error" : ""
+              }`}
+              name="password"
+              placeholder="Password"
+              required
+              value={password}
+              onChange={handlePasswordChange}
+            />
+          </label>
+        </div>
+      </div>
+    </ModalWithForm>
```

Comment:
**Could be improved**

If it’s interesting for you here is a way to make a special wrapper for any `modal` with cross button, `Escape` and `overlay` listeners inside. You should create component `Modal`, store it in `Modal.js`. 


```jsx

import { useEffect } from "react";

export const Modal = ({ name, onClose, children }) => {
  // here is `useEffect` for the `Escape` listener
  useEffect(() => {
    // we should define the handler inside `useEffect`, so that it wouldn’t lose the reference to be able to remove it
    const handleEscape = (e) => {
      if (e.key === "Escape") {
        onClose();
      }
    };

    document.addEventListener("keydown", handleEscape);
    // don’t forget to remove the listener in the `clean-up` function
    return () => document.removeEventListener("keydown", handleEscape);
  }, [onClose]);

  // here is the overlay handler
  const handleOverlay = (e) => {
    if (e.target === e.currentTarget) {
      onClose();
    }
  };

  // then we add the main wrapper with class `modal`
  return (
    <div className={`modal modal_type_${name}`} onClick={handleOverlay}>
      {/* the container for the contents */}
      <div className="modal__container">
        {/* here will be anything you add as `children`*/}
        {children}
        {/* add the close button */}
        <button className="modal__close" type="button" onClick={onClose} />
      </div>
    </div>
  );
};



```


And now you can use `Modal` for any popup in the project:  `ItemModal`  and `ModalWithForm` 

```jsx


function ModalWithForm({ name, onClose, ...props}) {
  return (
    <Modal name={name} onClose={onClose}>
        <h2 className='popup__title'>{props.title}</h2>


```

You can make also `Input` for inputs and `Form` for any form in the project.